pkg/api/api.go

```diff
@@ -97,6 +115,7 @@ type API struct {
 	sourceIPs            *middleware.SourceIPExtractor
 	indexPage            *IndexPageContent
 	HTTPHeaderMiddleware *HTTPHeaderMiddleware
+	CORSOrigin           *regexp.Regexp
 }
 
 func New(cfg Config, serverCfg server.Config, s *server.Server, logger log.Logger) (*API, error) {
```

Comment:
I think we can validate the `corsRegexString` and complie the regex here

Reply:
what do you mean by this?